src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,295 @@
+                end
+                # Store the cartesion coordinates of intersection point
+                intr_list[counter] = int_pt
+                # Store which edge of poly_b the intersection point lies on
+                b_idx_list[counter] = jj
+                # Store the alpha values
+                alpha_a_list[counter] = alphas[1]
+                alpha_b_list[counter] = alphas[2]
+
+                counter = counter + 1
+            else
+                continue
+            end
+        end
+
+        # After iterating through all edges of poly_b for edge ii of poly_a,
+        # add the intersection points to a_list in CORRECT ORDER if we found any
+        if prev_counter < counter
+            # Order intersection points based on alpha values
+            new_order = sortperm(alpha_a_list[prev_counter:counter-1])
+            pts_to_add = Array{PolyNode, 1}(undef, counter - prev_counter)
+            for kk in eachindex(new_order)
+                # Create PolyNodes of the new intersection points in the correct order
+                # and store the correct index in a_idx_list
+                pts_to_add[new_order[kk]] = PolyNode(prev_counter+kk-1, true, 0, false, alpha_a_list[prev_counter+kk-1])
+                a_idx_list[prev_counter+kk-1] = acount + new_order[kk] - 1;
+            end
+
+            # Add the PolyNodes to a_list and update acount
+            splice!(a_list, acount:(acount-1), pts_to_add)
+            acount = acount + counter - prev_counter
+            
+        end
+        
+    end
+
+    # Truncate the inter_related lists
+    intr_list = intr_list[1:counter-1] 
+    a_idx_list = a_idx_list[1:counter-1]
+    b_idx_list = b_idx_list[1:counter-1]
+    alpha_a_list = alpha_a_list[1:counter-1]
+    alpha_b_list = alpha_b_list[1:counter-1]
+    return intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list, counter
+end
+
+"""
+    _build_b_list(
+        b_list, intr_list, a_idx_list, 
+        b_idx_list, alpha_b_list, counter, k
+    )::a_list, b_list
+
+    This function builds b_list. Note that after calling this function, b_list
+    is not fully updated. The entry/exit flags still need to be updated.
+"""
+
+function _build_b_list(b_list, intr_list, a_idx_list, b_idx_list, alpha_b_list, counter, k)
+    # Iterate through the b_list and add in intersection points
+    # Occasionally I need to skip the new points I added to the array
+    skip = false
+    num_skips = 0
+    b_neighbors = Array{Int, 1}(undef, k)
+    for ii in 1:(length(b_list)+length(intr_list))
+        # TODO: this skipping scheme could be made nicer
+        if skip
+            num_skips = num_skips - 1
+            if num_skips == 0
+                skip = false
+            end
+            continue
+        end
+        # Find the index in the intr_list (same as b_idx_list) where the inter point is
+        # TODO:find all is inefficient though, so it might be better to make dictionary of indices
```

Comment:
Let's collect these TODOs and put them in an issue. We can call and talk about how to do that if you don't know. Since they aren't getting done pre-merge it probably makes more sense to put them in an issue rather than to leave  them lingering in the code.

Reply:
Do I make an issue for my TODOs on this fork or should I do it on the original GeometryOps repo after I open a pull request there?

Reply:
On the fork!